DM-54645: Add garbage collection metrics to task metadata (#562)
lsst/pipe_base: doc/changes/DM-54645.misc.md

> SingleQuantumExecutor adds garbage collection metrics to task metadata under "quantum.gc_metrics" key.

Reviewer: ```suggestion
Added garbage collection metrics to `SingleQuantumExecutor` with metrics stored in task metadata under `quantum.gc_metrics` key.
```
Changing tense.
